pandas/core/apply.py

```diff
@@ -346,14 +346,21 @@ def apply_series_generator(self) -> Tuple[ResType, "Index"]:
     def wrap_results(
         self, results: ResType, res_index: "Index"
     ) -> Union["Series", "DataFrame"]:
+        from pandas import Series
 
         # see if we can infer the results
         if len(results) > 0 and 0 in results and is_sequence(results[0]):
 
             return self.wrap_results_for_axis(results, res_index)
 
         # dict of scalars
-        result = self.obj._constructor_sliced(results)
+        # TODO: Remove if/else block when default dtype of Series is changed to object
```

Comment:
Will changing this to object dtype in the future result in a breaking change? What user-level code actually hits this?

Reply:
The comment seems outdated. I needed to pass `np.float64` for the `is_empty ` case to make the tests pass. Should I just remove the comment or do you have additional concerns?

Reply:
I have to break the tests to see which user-level code hits this. I'll add it as a comment as well

Reply:
Otherwise, this raises FutureWarnings in:
`pandas/tests/frame/test_analytics.py::TestDataFrameAnalytics::test_mean_excludes_datetimes`. The code is (raised on `df.mean()`):
```
tz = None
df = pd.DataFrame({"A": [pd.Timestamp("2000", tz=tz)] * 2})
result = df.mean()
```
where result is asserted against `pd.Series(dtype=np.float64)`

**Note:** The same happens when `tz = "UTC"`